R/drive_path.R

```diff
@@ -30,7 +30,13 @@ get_leaf <- function(path = NULL) {
   ## guarantee: we have found at least one item with correct name for
   ## each piece of path
 
-  hits$depth <- match(hits$name, path_pieces)
```

Comment:
Yeah, this is *alleged* depth, not actual. But then below I need to use the information when vetting the paths-to-root. Currently I'm not doing so, which leads to the false positive you discovered. ☹️ I will probably make a separate PR with an alternative fix.